platform/linux-generic/odp_ipsec.c

```diff
@@ -798,6 +800,12 @@ static ipsec_sa_t *ipsec_in_single(odp_packet_t pkt,
 		odp_packet_parse(pkt, state.ip_offset, &parse_param);
 	}
 
+	goto exit;
+
+post_ipsec_err_cnt_update:
```

Comment:
If _odp_ipsec_sa_stats_update() (or _odp_ipsec_sa_replay_update() before the antireplay check ordering fix is included) fails, we should come here to not count the packets as successes.

This whole mechanism would also be needed in outbound processing. There can be errors after lifetime updating there too.

Reply:
> If _odp_ipsec_sa_stats_update() (or _odp_ipsec_sa_replay_update() before the antireplay check ordering fix is included) fails, we should come here to not count the packets as successes.

Correct. Actually I was thinking that I would rebase after the other PR is merged. So missed this part.

> 
> This whole mechanism would also be needed in outbound processing. There can be errors after lifetime updating there too.

Agreed. Will add this in next version.

Do you agree with the change in labels? (Parent patch to this change). Outbound also would need a similar adjustment. IMO, calling the label 'err' and success also taking that code path may not look right.

Reply:
> Do you agree with the change in labels?

I do not think I would have made the change. I do not think it is a problem that the success path falls through to the same code that comes after the err label. Only the error cases actually use the label. If the label were already named exit, I would not change it either.

So, I would slightly prefer not changing the label, but it is a matter of taste and if you want to change it I am ok with it (which is why I did not comment it in the review).

Reply:
Understood. Without the label rename, my code change in case of success would have had like,
```
	goto err;

post_ipsec_err_cnt_update:
```

Or duplicate what is being done as part of err(/exit). Let me recheck if similar kind of change is required in the out function as well.

Reply:
Right, now I see it. I was looking at just that patch when making the previous comment, but after the latter patch the exit label makes more sense.